documentation/src/main/asciidoc/topics/upgrading.adoc

```diff
@@ -2,9 +2,12 @@
 
 == Encoding
 
-Conversions of cache data between `application/x-java-object` and `application/json` are deprecated and will be removed
+* Conversions of cache data between `application/x-java-object` and `application/json` are deprecated and will be removed
```

Comment:
I must have missed this part. Why would we preclude someone doing this if they have a transcoder that would support it?

Reply:
+1 for the deprecation, `application/x-java-object` on the server is a really niche case. It was probably useful to support compatibility between the different servers (and embedded back) in the day, but nowadays it's much better to use `application/protostream` for that purpose.

The only thing I'm not sure about is how good the protostream <-> JSON transcoding is, considering that protostream doesn't have map support yet, and JSON is all about mixing maps with static keys (i.e. object field names) and maps with dynamic keys.

Reply:
We map support comes, we'll probably follow the spec:  https://developers.google.com/protocol-buffers/docs/proto3#json